src/App.tsx

```diff
@@ -1,73 +1,125 @@
-import React from 'react';
+import React, { useState, useRef, useEffect } from 'react';
 import './App.scss';
 import { peopleFromServer } from './data/people';
+import classNames from 'classnames';
+import { Person } from './types/Person';
+import { useDebounce } from './hooks/useDebounce';
 
-export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+interface AppProps {
+  delay?: number;
+  onSelected?: (person: Person | null) => void;
+}
+
+export const App: React.FC<AppProps> = ({ delay = 300, onSelected }) => {
+  const [inputFocused, setInputFocused] = useState(false);
+  const [selectedPerson, setSelectedPerson] = useState<Person | null>(null);
+  const [searchQuery, setSearchQuery] = useState('');
+  const dropdownRef = useRef<HTMLDivElement | null>(null);
+
+  const debouncedQuery = useDebounce(searchQuery, delay);
+
+  const handleQueryChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setSearchQuery(event.target.value);
+    setSelectedPerson(null);
+  };
+
+  const handlePersonSelect = (newTarget: Person) => {
+    setSelectedPerson(newTarget);
+    setInputFocused(false);
+    onSelected?.(newTarget); // Pass the selected person
+  };
+
+  // Close dropdown when clicking outside
+  useEffect(() => {
+    const handleClickOutside = (event: MouseEvent) => {
+      if (
+        dropdownRef.current &&
+        !dropdownRef.current.contains(event.target as Node)
+      ) {
+        setInputFocused(false);
+      }
+    };
+
+    const dropdownNode = dropdownRef.current;
+
+    dropdownNode?.addEventListener('mousedown', handleClickOutside);
+
+    return () => {
+      dropdownNode?.removeEventListener('mousedown', handleClickOutside);
```

Comment:
Direct DOM manipulation using `addEventListener` and `removeEventListener` is discouraged in React. Consider using React's event handling mechanisms or hooks to manage this interaction.